recipe-server/normandy/recipes/tests/__init__.py

```diff
@@ -49,7 +49,8 @@ class Meta:
 
     @factory.lazy_attribute
     def implementation_hash(action):
-        return sri_hash(action.implementation.encode(), url_safe=True)
+        if action.implementation is not None:
+            return sri_hash(action.implementation.encode(), url_safe=True)
```

Comment:
Perhaps you can take a look at this. Perhaps I don't understand how this `@factory.lazy_attribute` works. Arguably, the method shouldn't even be called/triggered for the actions without an implementation.